test/verifier/embedded-zkp-verifier.test.ts

```diff
@@ -70,6 +75,83 @@ describe("Embedded ZKP Verifier", function () {
     expect(proofStatus.blockTimestamp).to.be.equal(txResTimestamp);
   });
 
+  it("test submit response v2", async () => {
+    const globalStateMessage = {
+      timestamp: BigInt(Math.floor(Date.now() / 1000)),
+      idType: "0x01A1",
+      root: 0n,
+      replacedAtTimestamp: 0n,
+    };
+
+    const identityStateMessage1 = {
+      timestamp: BigInt(Math.floor(Date.now() / 1000)),
+      id: 25530185136167283063987925153802803371825564143650291260157676786685420033n,
+      state: 4595702004868323299100310062178085028712435650290319955390778053863052230284n,
+      replacedAtTimestamp: 0n,
+    };
+
+    const identityStateUpdate2 = {
+      timestamp: BigInt(Math.floor(Date.now() / 1000)),
+      id: 25530185136167283063987925153802803371825564143650291260157676786685420033n,
+      state: 16775015541053109108201708100382933592407720757224325883910784163897594100403n,
+      replacedAtTimestamp: 1724858009n,
+    };
+
+    await verifier.setZKPRequest(0, {
+      metadata: "metadata",
+      validator: await sig.getAddress(),
+      data: packValidatorParams(query),
+    });
+
+    const { inputs, pi_a, pi_b, pi_c } = prepareInputs(proofJson);
+
+    const zkProof = packZKProof(inputs, pi_a, pi_b, pi_c);
+    const [signer] = await ethers.getSigners();
+    const crossChainProofs = packCrossChainProofs([
+      {
+        proofType: "globalStateProof",
+        proof: await packGlobalStateUpdateWithSignature(globalStateMessage, signer),
+      },
+      {
+        proofType: "stateProof",
+        proof: await packIdentityStateUpdateWithSignature(identityStateMessage1, signer),
+      },
+      {
+        proofType: "stateProof",
+        proof: await packIdentityStateUpdateWithSignature(identityStateUpdate2, signer),
+      },
+    ]);
```

Comment:
If not mistaken, this code repeats in many placed of the repo. Consider replacing with some helper function

Reply:
I created a helper function to build the proofs.